### PR TITLE
Updater

### LINKED ISSUE #239: Updater Coremod
Uses asar and GitHub releases

### PR DESCRIPTION
**Pull request by asportnoy**
Implements the plugin/theme updater! To use it, run:
```js
await replugged.updater.checkAllUpdates()
await replugged.updater.installAllUpdates()
```
Defaults to limiting checks to every 15 minutes. No auto updating yet. Note that the GitHub API has a limit of 60 requests per hour when unauthenticated, so be careful with how often you run it.

Will need updates in the future to work with the UI (once it exists) and work with the store (once it also exists). Some of the code could probably be shared with the plugin/theme installer (once it exists too), so that's cool.

Closes #239

### CODE REVIEW

File: src/main/ipc/updater.ts
Comment (E-boi) [inline src/main/ipc/updater.ts:63]: Instead of looking for if a file has .asar it should look for the filename because ideally the filename will be plugin/theme id. Also this makes it so monorepos wont work like [mine](https://github.com/E-boi/replugged-plugins).

**Comment by asportnoy** [comment]
Pointing to `better-settings` for now for review purposes, but will not merge until that PR is merged

File: src/main/ipc/updater.ts
Comment (asportnoy) [inline src/main/ipc/updater.ts:22]: todo: make this concurrent

File: src/renderer/managers/updater.ts
Comment (asportnoy) [inline src/renderer/managers/updater.ts:79]: todo: warn

File: src/renderer/managers/updater.ts
Comment (asportnoy) [inline src/renderer/managers/updater.ts:72]: todo: warn

File: src/renderer/managers/updater.ts
Comment (asportnoy) [inline src/renderer/managers/updater.ts:180]: todo: do before reloading

**Review by FifiTheBulldog — COMMENTED** [review]
I'm very hesitant to approve this at this stage for a few reasons:

- So much of it is going to need to be changed for the store.
- I'm not sure how reliable this is; in my testing, this appears to install updates, but it fails to reload them, throwing bizarre syntax errors instead. (There's a systemic problem here that renders the reload function useless anyway as long as plugins are shipped as asars.)
- I'm working on a major update for the plugin manager that will probably include some breaking changes for this.

File: src/renderer/managers/plugins.ts
Comment (asportnoy) [reply src/renderer/managers/plugins.ts:157]: Adds consistency with the theme manager since that requires `.asar` for asars

**Comment by asportnoy** [comment]
@FifiTheBulldog updated to work with the new plugin manager. GitHub will be an update source on top of the store. I designed it so it can be extended for the store once it's out.

**Review by FifiTheBulldog — APPROVED** [review]
Merge whenever you're ready

**Pull request merged**
